### PR TITLE
move section logic out of Tiles and to responsibility of caller

### PR DESCRIPTION
**Pull request by nickjvm**

### CODE REVIEW

**Comment by alansouzati** [comment]
CI failed because of this:

`20:32  error  Unexpected trailing comma  comma-dangle`

File: src/js/components/Tiles.js [hunk 1]
```diff
@@ -112,79 +112,33 @@ export default class IndexTiles extends Component {
...
-          }
-          tiles.push(this._renderTile(item));
-        } else {
-          // done
-          break;
-        }
-      }
+    result.sections.forEach((section, i) => {
+      const { actions, items } = section;
+      const tiles = items.map(item=>{
```
Comment (ericsoderberghp) [inline src/js/components/Tiles.js]: Our convention is to have space characters on either side of the '=>'.

File: src/js/components/Tiles.js [hunk 2]
```diff
@@ -112,79 +112,33 @@ export default class IndexTiles extends Component {
...
-            selectionIndex = tiles.length;
-          }
-          tiles.push(this._renderTile(item));
-        } else {
-          // done
-          break;
-        }
-      }
+    result.sections.forEach((section, i) => {
+      const { actions, items } = section;
```
Comment (ericsoderberghp) [inline src/js/components/Tiles.js]: Could put `label` here too.

**Comment by ericsoderberghp** [comment]
Nicely done. One more request. We try to allow consumers some time to migrate their code. Can you preserve the old behavior but add a console warning that putting `sections` in attributes is deprecated in favor of putting it in the result object.

Also, the term 'result' doesn't seem right to me, as it seems more oriented towards a request/reply context. I'm wondering if something more along the lines of 'window', 'snapshot', 'data', or perhaps 'index' would be better. Any ideas?

**Comment by nickjvm** [comment]
Makes sense; I'll add the previous functionality back with a warning. 

As for the prop name - I think 'data' makes more sense here. I'll make that change and also add a warning when using `result` prop.

**Comment by ericsoderberghp** [comment]
Nicely done. Thanks.

**Pull request merged**